jujugui/static/gui/src/app/components/inspector-config/inspector-config.js

```diff
@@ -0,0 +1,166 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('inspector-config', function() {
+
+  juju.components.Configuration = React.createClass({
+
+    _importConfig: function() {},
+    _resetValues: function() {},
+
+    /**
+      Callback handler for clicking the Save config button.
+
+      @method _saveConfig
+    */
+    _saveConfig: function() {
+      var refs = this.refs;
+      var configValues = {};
+      Object.keys(refs).forEach((ref) => {
+        // Just in case we ever have any sub components which have refs
+        // and aren't a configuration component.
+        var isConfig = ref.split('-')[0] === 'Config';
+        if (isConfig) {
+          var value;
+          if (refs[ref].state) {
```

Comment:
You could probably store `refs[ref]` in a variable instead of looking it up a number of times here.